references/classification/train.py

```diff
@@ -11,26 +11,38 @@
 
 import utils
 
+try:
```

Comment:
Can we move this import later on, only if the user pass the `--apex` flag?

Reply:
the import should be at module level. I'm not sure how we can check the `--apex` flag before entering `main`.

Reply:
In this case, I'd prefer to do it slightly differently.

Currently, we are unconditionally printing `Please install apex from https://www.github.com/nvidia/apex to enable mixed-precision training.`, even if the user has not requested it.

What about the following:

```python
try:
    from apex import amp, optimizers
except ImportError:
    amp, optimizers = None, None
```

and then, later on in the code, if the user passes `--apex`, we check that both `amp` and `optimizers` is not `None`, and if that's not the case raise an error?

Reply:
perfect. Let's go with this.

Reply:
currently not using "optimizers". Removing.